bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Trajectory/State.cpp

```diff
@@ -49,7 +49,16 @@ inline void OpenSpaceToolkitAstrodynamicsPy_Trajectory_State(pybind11::module& a
         .def("get_coordinates_subsets", &State::getCoordinatesSubsets)
         .def("get_frame", &State::getFrame)
 
-        .def("extract_coordinates", &State::extractCoordinates, arg("coordinates_subset"))
+        .def(
+            "extract_coordinates",
+            overload_cast<const Shared<const CoordinatesSubset>&>(&State::extractCoordinates, const_),
```

Comment:
What does this casting do? And the `const_` is a keyword that pybind recognizes?

Reply:
The cast is needed because Python has no concept of function overloading like in strongly typed languages (C++/Java). This cast is just a PyBind construct to let it know that it needs to do some magic in the backend to handle that.

`const` is to indicate that the C++ function is `const`. Yeah it's just another PyBind construct as well.

Reply:
(`const` for a class function just means that it's not allowed to modify any of the class attributes)
https://www.geeksforgeeks.org/const-member-functions-c/